react/useAddressRules.tsx

```diff
@@ -1,15 +1,34 @@
 import { useMemo } from 'react'
 import { useQuery } from 'react-apollo'
+import {
+  CountryData,
+  DisplayData,
+  AddressFieldsData,
+  LocationSelectData,
+} from 'vtex.country-data-settings'
 
 import addressRulesQuery from './graphql/addressRulesQuery.gql'
 
+interface RulesQuery {
+  allCountriesData: CountryData[]
+}
+
 const useAddressRules = () => {
-  const { data } = useQuery(addressRulesQuery)
+  const { data } = useQuery<RulesQuery>(addressRulesQuery)
```

Comment:
Wdyt of:
```suggestion
  const { data } = useQuery<Query>(addressRulesQuery)
```

Where `Query` comes from `vtex.country-data-settings` import, so it can stay up to date if something changes

Reply:
but I'm not querying for all the things `vtex.country-data-settings` exports, with your suggestion I could access `data.countryData.addressFields` and typescript would allow it but in reality it throw an error bc I `countryData` isn't in my query. I'm not a big fan of typing the way I did either, because typescript thinks I'm querying for all the fields the `allCountriesData` exposes but that's not true, ideally I would just explicitly type for only the fields I have in the query but that's too much work haha

Reply:
> but I'm not querying for all the things vtex.country-data-settings exports, with your suggestion I could access data.countryData.addressFields and typescript would allow it but in reality it throw an error bc I countryData isn't in my query.

fair enough

>  I'm not a big fan of typing the way I did either, because typescript thinks I'm querying for all the fields the allCountriesData exposes but that's not true

tru

> ideally I would just explicitly type for only the fields I have in the query but that's too much work haha

wait, shouldn't this be generated as well? I've never realized that this was a problem, but now that you said it...

Reply:
> wait, shouldn't this be generated as well? I've never realized that this was a problem, but now that you said it...

it should totally be generated, and this was the source of many discussions on slack 😅

Reply:
oh, and what was the outcome? If nothing will be done about this, do you think it would be worth it to do that ourselves? It doesn't _look_ like something hard to implement and seems to pay off in the long run

Reply:
I tried to implement this in toolbelt but it didn't go forward, the team at the time suggested to implement it on builder-hub instead but it was way out of my scope :-(

Reply:
interesting, maybe to do that on builder-hub we just have to include `react/graphql` in this line: https://github.com/vtex/builder-hub/blob/master/node/build/utils/publicAppTypes/graphqlTypes.ts#L33

but thats just a very naive guess 👀

Reply:
but sure, that can be addressed later on